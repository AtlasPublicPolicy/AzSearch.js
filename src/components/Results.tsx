import * as React from "react";
import { PropsType } from "../containers/ResultsContainer";
import * as objAssign from "object-assign";

export type State = {};

class Results extends React.Component<PropsType, State> {
    render() {
        const { results, template } = this.props;

        const renderedResults = results.map((result: any, index: number) => {
            let html = template ? template.render(result) : null;
            if (html) {
                return (
                    <div className={"searchResults__result col-xs-12 col-sm-12"} key={index} dangerouslySetInnerHTML={{ __html: html }} />
                );
            }
            else {
                return (
                    <div className={"searchResults__result col-xs-12 col-sm-12"} key={index} >
                        <pre>
                            <code>
                                {JSON.stringify(result, null, 4)}
                            </code>
                        </pre>
                    </div>
                );
            }
        });

        return (
            <div>
                {renderedResults}
            </div>
        );
    }
}

export default Results;