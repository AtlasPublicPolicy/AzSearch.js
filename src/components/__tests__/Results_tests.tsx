import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import Results from "../Results";

describe("Dummy snapshot test for <Results/>", () => {
    it("renders", () => {
        let results = [
            {
                key: "foo",
                text: "bar"
            },
            {
                key: "fizz",
                text: "buzz"
            }
        ];
        expect(TestUtils.createRenderer().render(
            <Results template={null} css={null} results={results} top={50} skip={0} count={results.length} />
        )).toMatchSnapshot();
    });
});