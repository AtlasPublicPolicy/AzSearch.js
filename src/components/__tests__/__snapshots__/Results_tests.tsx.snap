// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders 1`] = `
<div>
  <div
    className="results__blurb row"
  >
    1
     - 
    2
     
     of 2
  </div>
  <div
    className="searchResults__result col-xs-12 col-sm-12"
  >
    <pre>
      <code>
        {
            "key": "foo",
            "text": "bar"
        }
      </code>
    </pre>
  </div>
  <div
    className="searchResults__result col-xs-12 col-sm-12"
  >
    <pre>
      <code>
        {
            "key": "fizz",
            "text": "buzz"
        }
      </code>
    </pre>
  </div>
</div>
`;
